daemon/oci_linux.go

```diff
@@ -720,16 +720,18 @@ func sysctlExists(s string) bool {
 // WithCommonOptions sets common docker options
 func WithCommonOptions(daemon *Daemon, c *container.Container) coci.SpecOpts {
 	return func(ctx context.Context, _ coci.Client, _ *containers.Container, s *coci.Spec) error {
-		if c.BaseFS == nil {
+		if c.BaseFS == nil && !daemon.usesSnapshotter() {
 			return errors.New("populateCommonSpec: BaseFS of container " + c.ID + " is unexpectedly nil")
 		}
 		linkedEnv, err := daemon.setupLinkedContainers(c)
 		if err != nil {
 			return err
 		}
-		s.Root = &specs.Root{
-			Path:     c.BaseFS.Path(),
-			Readonly: c.HostConfig.ReadonlyRootfs,
+		if !daemon.usesSnapshotter() {
+			s.Root = &specs.Root{
```

Comment:
I see there's various locations where `s.Root` is referenced; should this be initialised with an empty `specs.Root` if `daemon.usesSnapshotter()` (to prevent panics), e.g. this may panic? https://github.com/moby/moby/blob/9ccedde9703bbef70881a0e8a3006310b7d3d788/libcontainerd/remote/client.go#L131


Or should the field be propagated in some other way? There's also the ReadOnly option that is discarded now; is that passed on through some other ways?

Reply:
enabling this flag in the current codebase **will break**, this PR just introduces the feature flag and utility method, with an initial illustration example where it is obviously needed, but it is not expected enabling this flag won't trigger a panic.